app/background.cc

```diff
@@ -183,11 +185,17 @@ int grab_background(std::shared_ptr<background_t> pbkd, int width, int height, c
     if (pbkd->video) {
         // grab frame & frame no. under mutex
         std::unique_lock<std::mutex> hold(pbkd->rawmux);
-        cv::resize(pbkd->raw, out, cv::Size(width, height));
+        cv::Rect crop = calcCropping(pbkd->raw.cols, pbkd->raw.rows, width, height);
+        cv::resize(pbkd->raw(crop), out, cv::Size(width, height));
         frm = pbkd->frame;
     } else {
         // resize still image as requested into out
-        cv::resize(pbkd->raw, out, cv::Size(width, height));
+        cv::Rect crop = calcCropping(pbkd->raw.cols, pbkd->raw.rows, width, height);
```

Comment:
..as above, could be removed in favour of a state variable, calculated just once.

Reply:
I must think about that.

Reply:
I must review this part.

Reply:
I have put the variable bg_stored into the structure background_t.
The code is now:
 ```
    } else {
		if (!pbkd->bg_stored) {
			// resize still image as requested into out
			cv::Rect crop = bs_calc_cropping(pbkd->raw.cols, pbkd->raw.rows, width, height);
			// Under some circumstances we must do the job in two steps!
			// Otherwise this resize(pbkd->raw(crop), pbkd->raw, ...) may fail.
			pbkd->raw(crop).copyTo(pbkd->raw);
			cv::resize(pbkd->raw, pbkd->raw, cv::Size(width, height));
			pbkd->bg_stored = true;
		}
		pbkd->raw = out;
		frm = 1;
	}
    return frm;
```

Reply:
Don't worry too much about this, it's a minor performance enhancement to avoid a couple of floating point ops per frame, when we are running a whole inference model too.. I'd rather the code was readable than maximum speed :smile: